### ISSUE TITLE
Don't reset Variables when autoReload triggers

### ISSUE THREAD

**Issue by pderuiter77** [NLog/NLog#1498]
This is a Feature request.
NLog version 4.3.4
Platform: net45

```
<nlog autoReload="true" throwExceptions="true" throwConfigExceptions="true" internalLogFile="nlog.txt" xmlns="http://www.nlog-project.org/schemas/NLog.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <variable name="Batch" value="" />
  <targets>
    <target name="allLog" xsi:type="File" fileName="Log Batch-${var:Batch}.txt" layout="${longdate}|${level:uppercase=true}|${logger}|${callsite}|${gdc:item=global}|${ndc:separator=->}|${message}" />
    <target name="errorLog" xsi:type="File" fileName="Errors Batch-${var:Batch}.txt" />
  </targets>
  <rules>
    <logger name="LogTest.Form1" writeTo="allLog" />
    <logger name="LogTest.Form1" minLevel="Error" writeTo="errorLog" />
  </rules>
</nlog>
```

Current configuration:

When i change the configuration file, the variables get reset, which means my filenames are off.

I request that on an autoreload, the variables (or at least the ones without values in the config) are taken from the current configuration

**Comment by 304NotModified** [comment]
That's a good idea :)

**Comment by 304NotModified** [comment]
Could you help us with an unit test? Would boost the implementation

**Comment by 304NotModified** [comment]
This is fixed in 4.4 (optional, default false)
